api/tests/identity-access-management/unit/domain/usecases/validate-user-account-email.usecase.test.js

```diff
@@ -0,0 +1,144 @@
+  let clock, now;
+
+  const token = '00000000-0000-0000-0000-000000000000';
+  const defaultRedirectionUrl = 'https://app.pix.fr/connexion';
+
+  beforeEach(function () {
+    emailValidationDemandRepository = {
+      get: sinon.stub(),
+      remove: sinon.stub(),
+    };
+    userRepository = {
+      get: sinon.stub(),
+      update: sinon.stub(),
+    };
+
+    now = new Date('2024-06-26');
+    clock = sinon.useFakeTimers({ now, toFake: ['Date'] });
+
+    sinon.stub(monitoringTools, 'logErrorWithCorrelationIds');
+  });
+
+  afterEach(function () {
+    clock.restore();
+  });
+
+  context('when token property is not given', function () {
+    it('returns default redirection URL', async function () {
+      // when
+      const redirectionUrl = await validateUserAccountEmail({
+        emailValidationDemandRepository,
+      });
+
+      // then
+      expect(emailValidationDemandRepository.get).to.not.have.been.called;
+      expect(emailValidationDemandRepository.remove).to.not.have.been.called;
+      expect(redirectionUrl).to.equal(defaultRedirectionUrl);
+    });
+
+    it('returns the given redirection URL', async function () {
+      // when
+      const redirectionUrl = await validateUserAccountEmail({
+        redirectUri: 'https://test.com',
+        emailValidationDemandRepository,
+      });
+
+      // then
+      expect(emailValidationDemandRepository.get).to.not.have.been.called;
+      expect(emailValidationDemandRepository.remove).to.not.have.been.called;
+      expect(redirectionUrl).to.equal('https://test.com');
+    });
+  });
+
+  context('when the email validation demand does not exist', function () {
+    it('returns default redirection URL', async function () {
+      // given
+      emailValidationDemandRepository.get.resolves(null);
+
+      // when
+      const redirectionUrl = await validateUserAccountEmail({
+        token,
+        emailValidationDemandRepository,
+      });
+
+      // then
+      expect(emailValidationDemandRepository.get).to.have.been.calledWith(token);
+      expect(emailValidationDemandRepository.remove).to.not.have.been.called;
+      expect(redirectionUrl).to.equal(defaultRedirectionUrl);
+    });
+
+    it('returns the given redirection URL', async function () {
+      // given
+      emailValidationDemandRepository.get.resolves(null);
+
+      // when
+      const redirectionUrl = await validateUserAccountEmail({
+        token,
+        redirectUri: 'https://test.com',
+        emailValidationDemandRepository,
+      });
+
+      // then
+      expect(emailValidationDemandRepository.get).to.have.been.calledWith(token);
+      expect(emailValidationDemandRepository.remove).to.not.have.been.called;
+      expect(redirectionUrl).to.equal('https://test.com');
+    });
+  });
+
+  context('when the email validation demand exists', function () {
+    it('validates user email, removes the demand and does not return', async function () {
```

Comment:
Préciser que ça retourne bien une redirectionUrl 😄 
```suggestion
    it('validates user email, removes the demand and returns a redirection url, async function () {
```